File: Assignment 7/Activity 3.js
function main() {
    // This program coverts the distance in miles into either the US system yards or metric system kilometers.
    var choice;
    
    choice = getChoice();
    if (choice == "Y" || choice == "y") {
        processUs();
    } else {
        if (choice == "N" || choice == "n") {
            processMetric();
        } else {
            window.alert("Please enter Y to use the US measurement system or N to use the  metric measurement system !");
        }
    }
}

function calculatecentimeters(miles) {
    var centimeters;
    
    centimeters = miles * 160934;
    
    return centimeters;
}

function calculatefeet(miles) {
    var feet;
    
    feet = miles * 5280;
    
    return feet;
}

function calculateinches(miles) {
    var inches;
    
    inches = miles * 63360;
    
    return inches;
}

function calculatekilometers(miles) {
    var kilometers;
    
    kilometers = miles * 1.60934;
    
    return kilometers;
}

function calculatemeters(miles) {
    var meters;
    
    meters = miles * 1609.34;
    
    return meters;
}

function calculateyards(miles) {
    var yards;
    
    yards = miles * 1760;
    
    return yards;
}

function getChoice() {
    var choice;
    
    window.alert("Would you like to use the US measurement system? Y or N ");
    choice = window.prompt('Enter a value for choice');
    
    return choice;
}

function getMiles() {
    var miles;
    
    window.alert("Enter a value for miles");
    miles = window.prompt('Enter a value for miles');
    
    return miles;
}

function metricDisplayResults(kilometers, meters, centimeters) {
    window.alert(kilometers.ToString() + " kilometers " + meters + " meters " + centimeters + " centimeter ");
}

function processMetric() {
    var miles;
    var kilometers;
    var meters;
    var centimeters;
    
    miles = getMiles();
    kilometers = calculatekilometers(miles);
    meters = calculatemeters(miles);
    centimeters = calculatecentimeters(miles);
    metricDisplayResults(kilometers, meters, centimeters);
}

function processUs() {
    var miles;
    var yards;
    var feet;
    var inches;
    
    miles = getMiles();
    yards = calculateyards(miles);
    feet = calculatefeet(miles);
    inches = calculateinches(miles);
    usDisplayResults(yards, feet, inches);
}

function usDisplayResults(yards, feet, inches) {
    window.alert(yards.ToString() + " yards " + feet + " feet " + inches + " inches ");
}

File: Assignment 7/Activity 3.py
# This program coverts the distance in miles 
# into either the US system yards or metric system kilometers.


def calculate_centimeters(miles):
    centimeters = miles * 160934
    return centimeters


def calculate_feet(miles):
    feet = miles * 5280
    return feet


def calculate_inches(miles):
    inches = miles * 63360
    return inches


def calculate_kilometers(miles):
    kilometers = miles * 1.60934
    return kilometers


def calculate_meters(miles):
    meters = miles * 1609.34
    return meters


def calculate_yards(miles):
    yards = miles * 1760
    return yards


def get_choice():
    print("Would you like to use the US measurement system or Metric ?")
    choice = input()
    return choice


def get_miles():
    print("Enter a value for miles")
    miles = float(input())
    return miles


def metric_display_results(kilometers, meters, centimeters):
    print(str(kilometers) + " kilometers " + 
    str(meters) + " meters " + str(centimeters) + " centimeter ")


def process_metric(miles):
    kilometers = calculate_kilometers(miles)
    meters = calculate_meters(miles)
    centimeters = calculate_centimeters(miles)
    metric_display_results(kilometers, meters, centimeters)


def process_us(miles):
    yards = calculate_yards(miles)
    feet = calculate_centimeters(miles)
    inches = calculate_inches(miles)
    us_display_results(yards, feet, inches)


def us_display_results(yards, feet, inches):
    print(str(yards) + " yards " + 
    str(feet) + " feet " + str(inches) + " inches ")


def main():
    miles = get_miles()
    choice = get_choice()
    if choice == "US" or choice == "Us":
        process_us(miles)
    else:
        if choice == "Metric" or choice == "metric":
            process_metric(miles)
        else:
            print("Please enter US or metric to use that metric measurement system !")


main()
